### PR TITLE
Generate SSL certificates with mkcert when available (self-signed otherwise)

### LINKED ISSUE #165: Better DX with SSL certificates with mkcert
Currently, SSL is available by default with auto-signed certificates. We already propose an alternative in [a cookbook about Mkcert](https://github.com/jolicode/docker-starter#how-to-generate-locally-trusted-ssl-certificates-with-mkcert).

I suggest that we no longer provide self-signed SSL certificates BUT a start of the infrastructure should run by default a modified version of the task `generate_certificates`.

- if certifcates are already generated, do nothing;
- else it should generate certificates with mkcert if the tool is available;
- else it should generate auto-signed certificates with the script in [infrastructure/docker/services/router/generate-ssl.sh](https://github.com/jolicode/docker-starter/blob/main/infrastructure/docker/services/router/generate-ssl.sh) (it could ask a confirmation before generating certificates but I prefer that it works out of the box without having to do anything).

### PR DESCRIPTION
**Pull request by pyrech**
Fix #165 

When certificates already exist:

image

When mkcert is available:

image

When mkcert is not available:

image

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -201,6 +203,45 @@ def help(c):
...
+                print('Please restart the infrastructure to use the new certificates with "inv up" or "inv start".')
+            return
+
+        c.run('infrastructure/docker/services/router/generate-ssl.sh')
+
+        print(Fore.GREEN + 'Successfully generated self-signed SSL certificates in infrastructure/docker/services/router/etc/ssl/certs/*.pem.')
+        print(Fore.YELLOW + 'Consider installing mkcert to generate locally trusted SSL certificates and run "inv generate-certificates --force".')
+        if force:
+            print('Please restart the infrastructure to use the new certificates with "inv up" or "inv start".')
+        return
```
Comment (lyrixx) [inline tasks.py:242]: Dead code

**Review by lyrixx — COMMENTED** [review]
Super cool

File: tasks.py [hunk 2]
```diff
@@ -201,6 +203,44 @@ def help(c):
...
+    with Builder(c):
+        if (os.path.isfile(c.root_dir + '/infrastructure/docker/services/router/etc/ssl/certs/cert.pem') and not force):
+            print(Fore.GREEN + 'SSL certificates found in infrastructure/docker/services/router/etc/ssl/certs/*.pem.')
+            print('Run "inv generate-certificates --force" to generate new certificates.')
+            return
+
+        if force:
+            if os.path.isfile(c.root_dir + '/infrastructure/docker/services/router/etc/ssl/certs/cert.pem'):
+                print('Removing existing certificates in infrastructure/docker/services/router/etc/ssl/certs/*.pem.')
+            os.remove('infrastructure/docker/services/router/etc/ssl/certs/cert.pem')
```
Comment (welcoMattic) [inline tasks.py:220]: it miss identation here, no?

File: tasks.py [hunk 3]
```diff
@@ -201,6 +203,44 @@ def help(c):
...
+            os.remove('infrastructure/docker/services/router/etc/ssl/certs/key.pem')
+
+        if find_executable('mkcert') is not None:
+            path_caroot = c.run('mkcert -CAROOT', hide=True).stdout.strip()
+
+            if not os.path.isdir(path_caroot):
+                print(Fore.RED + 'You must have mkcert CA Root installed on your host with `mkcert -install` command')
+                return
+
+            c.run('mkcert -cert-file infrastructure/docker/services/router/etc/ssl/certs/cert.pem -key-file infrastructure/docker/services/router/etc/ssl/certs/key.pem %s "*.%s"' % (c.root_domain, c.root_domain))
```
Comment (welcoMattic) [inline tasks.py:230]: it miss extra_domains here

**Review by welcoMattic — APPROVED** [review]

**Pull request merged**